I'm having problems to compile the following lines:

~~~c++
void CompositeCondition::EquationIdVector( EquationIdVectorType& rResult, 
                                          const ProcessInfo& rCurrentProcessInfo ) const
{

  rResult.resize(0,false);

  EquationIdVectorType LocalResult;

  for (ConditionIterator cn = mChildConditions.begin() ; cn != mChildConditions.end(); ++cn)
    {
      if( IsActive(cn, rCurrentProcessInfo) ){

	cn->EquationIdVector(LocalResult,rCurrentProcessInfo);

	for(unsigned int i=0; i<LocalResult.size(); i++)
	  {
	    rResult.push_back(LocalResult[i]);
	  }
      }

    }

}
~~~

It says:

D:\KRATOS_MULTIPHYSICS\Kratos\applications\DelaunayMeshingApplication\custom_conditions\composite_condition.cpp(229):
error C2440: 'inicializando': no se puede realizar la conversión de 'boost::iterators::indirect_iterator<std::_Vector_
const_iterator<std::_Vector_val<std::_Simple_types<_Ty>>>,boost::iterators::use_default,boost::iterators::use_default,
boost::iterators::use_default,boost::iterators::use_default>' a 'boost::iterators::indirect_iterator<std::_Vector_iter
ator<std::_Vector_val<std::_Simple_types<_Ty>>>,boost::iterators::use_default,boost::iterators::use_default,boost::ite
rators::use_default,boost::iterators::use_default>' [D:\KRATOS_MULTIPHYSICS\Kratos\build\Release\applications\Delaunay
MeshingApplication\KratosDelaunayMeshingCore.vcxproj]
          with
          [
              _Ty=Kratos::Condition::Pointer
          ]
  D:\KRATOS_MULTIPHYSICS\Kratos\applications\DelaunayMeshingApplication\custom_conditions\composite_condition.cpp(229)
  : note: Ningún constructor ha podido aceptar el tipo de origen o bien la resolución de sobrecarga del constructor er
  a ambigua
D:\KRATOS_MULTIPHYSICS\Kratos\applications\DelaunayMeshingApplication\custom_conditions\composite_condition.cpp(231):
error C2662: 'bool Kratos::CompositeCondition::IsActive(Kratos::CompositeCondition::ConditionIterator,const Kratos::Pr
ocessInfo &)': no se puede convertir el puntero 'this' de 'const Kratos::CompositeCondition' a 'Kratos::CompositeCondi
tion &' [D:\KRATOS_MULTIPHYSICS\Kratos\build\Release\applications\DelaunayMeshingApplication\KratosDelaunayMeshingCore
.vcxproj]

@philbucher @RiccardoRossi

--- a/applications/DelaunayMeshingApplication/custom_conditions/composite_condition.cpp
+++ b/applications/DelaunayMeshingApplication/custom_conditions/composite_condition.cpp
@@ -222,23 +222,23 @@ void CompositeCondition::EquationIdVector( EquationIdVectorType& rResult,
                                           const ProcessInfo& rCurrentProcessInfo ) const
 {
 
-  // rResult.resize(0,false);
+  rResult.resize(0,false);
 
-  // EquationIdVectorType LocalResult;
+  EquationIdVectorType LocalResult;
 
-  // for (ConditionIterator cn = mChildConditions.begin() ; cn != mChildConditions.end(); ++cn)
-  //   {
-  //     if( IsActive(cn, rCurrentProcessInfo) ){
+  for (ConditionIterator cn = mChildConditions.begin() ; cn != mChildConditions.end(); ++cn)
+    {
+      if( IsActive(cn, rCurrentProcessInfo) ){
 
-	// cn->EquationIdVector(LocalResult,rCurrentProcessInfo);
+	cn->EquationIdVector(LocalResult,rCurrentProcessInfo);
 
-	// for(unsigned int i=0; i<LocalResult.size(); i++)
-	//   {
-	//     rResult.push_back(LocalResult[i]);
-	//   }
-  //     }
+	for(unsigned int i=0; i<LocalResult.size(); i++)
+	  {
+	    rResult.push_back(LocalResult[i]);
+	  }
+      }
 
-  //   }
+    }
 
 }
 

--- a/applications/DelaunayMeshingApplication/custom_conditions/composite_condition.cpp
+++ b/applications/DelaunayMeshingApplication/custom_conditions/composite_condition.cpp
@@ -246,7 +246,7 @@ void CompositeCondition::EquationIdVector( EquationIdVectorType& rResult,
 //*********************************SET VALUE TO CHILDREN******************************
 //************************************************************************************
 
-bool CompositeCondition::IsActive( ConditionIterator iChildCondition, const ProcessInfo& rCurrentProcessInfo ) const
+bool CompositeCondition::IsActive( const ConditionIterator iChildCondition, const ProcessInfo& rCurrentProcessInfo ) const
 {
   if(rCurrentProcessInfo.Is(THERMAL) && iChildCondition->Is(THERMAL))
     return true;

--- a/applications/DelaunayMeshingApplication/custom_conditions/composite_condition.cpp
+++ b/applications/DelaunayMeshingApplication/custom_conditions/composite_condition.cpp
@@ -218,7 +218,7 @@ void CompositeCondition::GetDofList( DofsVectorType& rConditionalDofList, Proces
 //************************************************************************************
 //************************************************************************************
 
-void CompositeCondition::EquationIdVector( EquationIdVectorType& rResult, 
+void CompositeCondition::EquationIdVector( EquationIdVectorType& rResult,
                                           const ProcessInfo& rCurrentProcessInfo ) const
 {
 
@@ -246,7 +246,7 @@ void CompositeCondition::EquationIdVector( EquationIdVectorType& rResult,
 //*********************************SET VALUE TO CHILDREN******************************
 //************************************************************************************
 
-bool CompositeCondition::IsActive( ConditionIterator iChildCondition, const ProcessInfo& rCurrentProcessInfo )
+bool CompositeCondition::IsActive( ConditionIterator iChildCondition, const ProcessInfo& rCurrentProcessInfo ) const
 {
   if(rCurrentProcessInfo.Is(THERMAL) && iChildCondition->Is(THERMAL))
     return true;

--- a/applications/DelaunayMeshingApplication/custom_conditions/composite_condition.hpp
+++ b/applications/DelaunayMeshingApplication/custom_conditions/composite_condition.hpp
@@ -472,7 +472,7 @@ private:
     void InitializeChildren();
 
     //check problem type definition and if coincides return active true
-    bool IsActive(ConditionIterator iChildCondition, const ProcessInfo& rCurrentProcessInfo) const;
+    bool IsActive(const ConditionIterator iChildCondition, const ProcessInfo& rCurrentProcessInfo) const;
 
     //set specific data value to condition children
     template<class TVariableType> void SetValueToChildren(const TVariableType& rThisVariable){

--- a/applications/DelaunayMeshingApplication/custom_conditions/composite_condition.hpp
+++ b/applications/DelaunayMeshingApplication/custom_conditions/composite_condition.hpp
@@ -472,7 +472,7 @@ private:
     void InitializeChildren();
 
     //check problem type definition and if coincides return active true
-    bool IsActive(ConditionIterator iChildCondition, const ProcessInfo& rCurrentProcessInfo);
+    bool IsActive(ConditionIterator iChildCondition, const ProcessInfo& rCurrentProcessInfo) const;
 
     //set specific data value to condition children
     template<class TVariableType> void SetValueToChildren(const TVariableType& rThisVariable){

